packages/cached/lib/src/templates/deletes_cache_method_template.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:cached/src/models/deletes_cache_method.dart';
+import 'package:cached/src/models/streamed_cache_method.dart';
+import 'package:cached/src/templates/all_params_template.dart';
+import 'package:cached/src/utils/utils.dart';
+
+class DeletesCacheMethodTemplate {
+  DeletesCacheMethodTemplate(
+    this.method,
+    this.streamedCacheMethods,
+  ) : paramsTemplate = AllParamsTemplate(method.params);
+
+  final DeletesCacheMethod method;
+  final AllParamsTemplate paramsTemplate;
+  final List<StreamedCacheMethod>? streamedCacheMethods;
+
+  String generateMethod() {
+    final asyncModifier = isFuture(method.returnType) ? 'async' : '';
+    final awaitIfNeeded = isFuture(method.returnType) ? 'await' : '';
+
+    return '''
+    @override
+    ${method.returnType} ${method.name}(${paramsTemplate.generateParams()}) $asyncModifier {
+      try {
+        final result = $awaitIfNeeded super.${method.name}(${paramsTemplate.generateParamsUsage()});
+
+        ${_generateClearMaps()}
+
+        return result;
+      } catch (e) {
+        rethrow;
+      }
+    }
+    ''';
```

Comment:
`try..catch` is redundant here. Result of this will be exactly the same: 

```suggestion
    ${method.returnType} ${method.name}(${paramsTemplate.generateParams()}) $asyncModifier {
        final result = $awaitIfNeeded super.${method.name}(${paramsTemplate.generateParamsUsage()});

        ${_generateClearMaps()}

        return result;
    }
    ''';
```